web/src/beta/components/Modal/index.tsx

```diff
@@ -1,54 +1,140 @@
   button1,
   button2,
+  children,
   isVisible,
   onClose,
-  children,
+  sidebarTabs,
 }) => {
-  const theme = useTheme();
+  const [TabsFields, setTabsFields] = useState<SwitchField<SidebarTab>[]>([]);
+
+  useEffect(() => {
+    if (sidebarTabs) {
+      const convertedFields: SwitchField<SidebarTab>[] = sidebarTabs.map((tab, index) => ({
+        active: index === 0,
+        ...tab,
+      }));
+      setTabsFields(convertedFields);
+    }
+  }, [sidebarTabs]);
+
+  const { handleActivate, fields: tabs } = useManageSwitchState({
+    fields: TabsFields,
+  });
+
+  const handleTabChange = useCallback(
+    (tabId: string) => {
+      handleActivate(tabId);
+    },
+    [handleActivate],
+  );
+
   return (
-    <Wrapper className={className} size={size} isVisible={isVisible} onClose={onClose}>
-      <Title size="h4" weight="bold" color={theme.general.content.strong}>
-        {title}
-      </Title>
-      {children}
-      <ButtonWrapper>
-        {button2}
-        {button1}
-      </ButtonWrapper>
+    <Wrapper
+      className={className}
+      size={size}
+      isVisible={isVisible}
+      title={title}
+      onClose={onClose}>
+      <InnerWrapper>
+        {tabs.length > 0 && (
+          <NavBarWrapper>
+            {tabs.map(tab => (
+              <Tab key={tab.id} isSelected={tab.active} onClick={() => handleTabChange(tab.id)}>
+                {tab.label}
+              </Tab>
+            ))}
+          </NavBarWrapper>
+        )}
+        <ContentWrapper>
+          {tabs && <Content>{tabs.find(tab => tab.active === true)?.content}</Content>}
+          <Content> {children}</Content>
+          <ButtonWrapper>
+            {tabs.find(tab => tab.active === true)?.tabButton1 ?? button1}
+            {tabs.find(tab => tab.active === true)?.tabButton2 ?? button2}
+          </ButtonWrapper>
+        </ContentWrapper>
+      </InnerWrapper>
     </Wrapper>
   );
 };
 
-const Title = styled(Text)`
-  text-align: center;
-  margin-bottom: 24px;
+const InnerWrapper = styled.div`
+  display: flex;
+`;
+
+const NavBarWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: 16px;
+  gap: 10px;
+  border-right: 1px solid #525252;
+`;
+
+const Tab = styled.button<{ isSelected?: boolean }>`
+  display: flex;
+  padding: 4px 8px;
+  border-radius: 4px;
+  background: ${({ isSelected }) => (isSelected ? "#393939" : "transparent")};
+  color: ${({ isSelected }) => (isSelected ? "#E0E0E0" : "#393939")};
+`;
+
+const ContentWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const Content = styled.div`
+  display: flex;
+  padding: 24px;
+  flex-direction: column;
+  gap: 20px;
+  align-self: stretch;
 `;
 
 const ButtonWrapper = styled.div`
   display: flex;
+  flex-direction: row;
+  align-self: stretch;
   justify-content: flex-end;
-  margin: 20px -10px -10px 0;
+  border-top: 1px solid #525252;
+  & > :first-child {
+    order: 2;
+  }
+  & > :last-child {
+    order: 1;
+  }
```

Comment:
I think this will be a little confusing to remember when using this Modal. Can you leave it as is please 🙇🏻